src/promptbase/bigbench/bigbench_cot.py

```diff
@@ -201,35 +136,48 @@ def do_completion_cot(bbh_test_path, cot_prompt_path, test_name, cot_results_pat
 def process_cot(test_name: str, api_type="chat"):
     _logger.info("Starting process_cot")
     if test_name == "all":
-        subjects = SUBJECTS
-    elif test_name in SUBJECTS:
+        subjects = BIGBENCH_SUBJECTS
+    elif test_name in BIGBENCH_SUBJECTS:
         subjects = [test_name]
     else:
         print(f"Invalid test name: {test_name}")
         exit(1)
 
+    bigbench_data_root = get_datasets_path() / "BigBench"
+    cot_prompts_dir = bigbench_data_root / "cot-prompts"
+    bbh_test_dir = bigbench_data_root / "bbh"
+    generations_dir = get_generations_path()
+
+    if not os.path.exists(cot_prompts_dir):
+        print(f"COT prompt directory {cot_prompts_dir} does not exist")
+        exit(1)
+    elif not os.path.exists(bbh_test_dir):
+        print(f"BBH test directory {bbh_test_dir} does not exist")
+        exit(1)
+
     print(f"Processing CoT for BigBench subjects: {subjects}")
 
     threads = []
     for subject in subjects:
-        bbh_test_path = os.path.join(bbh_test_dir, f"{subject}.json")
-        cot_prompt_path = os.path.join(cot_prompts_dir, f"{subject}.txt")
-        # check if they exist
+        bbh_test_path = bbh_test_dir / f"{subject}.json"
+        cot_prompt_path = cot_prompts_dir / f"{subject}.txt"
         if not os.path.exists(bbh_test_path):
             print(f"Data file {bbh_test_path} does not exist")
         elif not os.path.exists(cot_prompt_path):
             print(f"COT prompt file {cot_prompt_path} does not exist")
 
         if api_type == "completion":
             _logger.info(f"Starting completion thread for {bbh_test_path}")
-            results_path = os.path.join(".", "results", "cot_results", "completion")
+            results_path = generations_dir / "bigbench" / "cot_results" / "completion"
+            os.makedirs(results_path, exist_ok=True)
```

Comment:
If `results_path` is a `pathlib.Path` object, there is a 'mkdir' (or something like that) method on it

Reply:
updated